### PR TITLE
Video refactoring

### PR DESCRIPTION
**Pull request by Hangary**
## Changes

Codes related to Video Container is modified and downloads container is added.

1. Refactored the original download codes and done a lot of debugging based on Suxiang's changes over video containers since the params and logic are changed a lot. Time Spent: 6 hrs.
2. Improved the download UI according to the professor's advice. Time spent: 5 hrs.
3. Refactored the video container and changed the button layout. Time spent: 5 hrs.
4. Debugged some potential issues over the video container. Time spent: 6 hrs.

## Examples

Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-11 at 04 52 08
Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-11 at 04 52 01
Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-11 at 04 51 57
Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-11 at 00 02 50

### CODE REVIEW

File: src/containers/DownloadContainer/DownloadContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.warn(e)
+      return null
+    }
+  }
+
+  const fetchLocalFiles = async () => {
+    let files = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory)
+    files = files.filter((fileName) => fileName.endsWith('.mp4'))
```
Comment (mattwalo32) [inline src/containers/DownloadContainer/DownloadContainer.jsx:36]: Will this grab all files that end with mp4? Even ones not related to CT

File: src/containers/DownloadContainer/DownloadContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    const videoItems = await Promise.all(
+      files.map(async (fileName) => {
+        const localUri = FileSystem.documentDirectory + fileName
+        const thumbnail = await generateThumbnail(localUri)
+        const filesize = await getFileSize(localUri)
+        return {
+          name: fileName.slice(0, -4),
```
Comment (mattwalo32) [inline src/containers/DownloadContainer/DownloadContainer.jsx:43]: Why -4?

File: src/containers/DownloadContainer/DownloadContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      })
+    )
+    setLocalVideos(videoItems)
+  }
+
+  useEffect(() => {
+    fetchLocalFiles()
+    const interval = setInterval(() => {
+      fetchLocalFiles()
+    }, 5000)
```
Comment (mattwalo32) [inline src/containers/DownloadContainer/DownloadContainer.jsx:59]: What's the reason for this periodic fetching

File: src/containers/DownloadContainer/DownloadContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    return () => {
+      clearInterval(interval)
+      // eslint-disable-next-line no-console
+      console.log('clear downloads interval')
```
Comment (mattwalo32) [inline src/containers/DownloadContainer/DownloadContainer.jsx:63]: Can you remove this log

**Review by mattwalo32 — APPROVED** [review]
This works great, very nice work!

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 3 above)
Comment (mattwalo32) [reply src/containers/DownloadContainer/DownloadContainer.jsx:59]: Is that for if someone starts a download then goes to the download screen before it finishes?

File: src/containers/VideoContainer/VideoContainer.js [hunk 5]
```diff
@@ -93,63 +176,85 @@ const VideoContainer = ({ videos, index }) => {
...
+                color: 'white',
+                alignSelf: 'center',
+                alignContent: 'center',
+                alignItems: 'center',
+              },
+              inputAndroid: {
+                height: '100%',
+                color: 'white',
+              },
+            }}
```
Comment (LunaMystic) [inline src/containers/VideoContainer/VideoContainer.js:225]: You might consider putting the style to style.js :)

**Review by LunaMystic — APPROVED** [review]
Nice work!

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 1 above)
Comment (Hangary) [reply src/containers/DownloadContainer/DownloadContainer.jsx:36]: `FileSystem.documentDirectory` is currently only used for storing downloaded videos. So we would not worry about this problem unless we decided to download other documents.

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 2 above)
Comment (Hangary) [reply src/containers/DownloadContainer/DownloadContainer.jsx:43]: This is to show the video name without the `.mp4` suffix.

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 3 above)
Comment (Hangary) [reply src/containers/DownloadContainer/DownloadContainer.jsx:59]: > Is that for if someone starts a download then goes to the download screen before it finishes?
Yes, exactly. This just checks the videos folder every 5 seconds.

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 1 above)
Comment (mattwalo32) [reply src/containers/DownloadContainer/DownloadContainer.jsx:36]: I guess I was more so asking if other applications can download to this directory. I.e. if I download a movie with something like netflix, is there a chance that we might accidentally pull that from the file?

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 2 above)
Comment (mattwalo32) [reply src/containers/DownloadContainer/DownloadContainer.jsx:43]: Could you make the suffix a variable and then call .length on it? I think that would make the code a bit more readable.

File: src/containers/DownloadContainer/DownloadContainer.jsx (hunk 1 above)
Comment (Hangary) [reply src/containers/DownloadContainer/DownloadContainer.jsx:36]: No, the expo filesystem would assign each app a unique document folder and this folder would only be accessed by this app.

**Pull request merged**